### ISSUE TITLE
Extension Host Terminated unexpectedly

### ISSUE THREAD

**Issue by tobiaslombacher** [microsoft/vscode#82499]
Issue Type: <b>Bug</b>

Update: I have found that if I run the same installation on a local administrator account the error does not happen. (Right Click -> Run as Admin) 

After upgrade from 1.37.1 to 1.38 or 1.39 of VS-Code the "extension host terminates unexpectedly". 

This happens also on a clean install without any modifications, extensions...

I have tried 

- new installation, with and without any extensions -> no difference
- new portable installation, with and without any extensions -> no difference
- disable git, since other suggest it may come from there -> no difference

The extensions I typically use are listed below.

This is the output from the error console:

> abstractExtensionService.ts:155 Extension host terminated unexpectedly. Code:  3221225477  Signal:  null
_onExtensionHostCrashed @ abstractExtensionService.ts:155
notificationsAlerts.ts:40 Extension host terminated unexpectedly.
onDidNotificationChange @ notificationsAlerts.ts:40

VS Code version: Code 1.39.1 (88f15d17dca836346e787762685a40bb5cce75a8, 2019-10-10T23:31:28.683Z)
OS version: Windows_NT x64 6.1.7601

<details>
<summary>System Info</summary>

|Item|Value|
|---|---|
|CPUs|Intel(R) Core(TM) i7-6700HQ CPU @ 2.60GHz (8 x 2592)|
|GPU Status|2d_canvas: enabled<br>flash_3d: enabled<br>flash_stage3d: enabled<br>flash_stage3d_baseline: enabled<br>gpu_compositing: enabled<br>multiple_raster_threads: enabled_on<br>native_gpu_memory_buffers: disabled_software<br>oop_rasterization: disabled_off<br>protected_video_decode: unavailable_off<br>rasterization: enabled<br>skia_deferred_display_list: disabled_off<br>skia_renderer: disabled_off<br>surface_synchronization: enabled_on<br>video_decode: enabled<br>viz_display_compositor: disabled_off<br>webgl: enabled<br>webgl2: enabled|
|Load (avg)|undefined|
|Memory (System)|31.88GB (22.94GB free)|
|Process Argv||
|Screen Reader|no|
|VM|44%|
</details><details><summary>Extensions (8)</summary>

Extension|Author (truncated)|Version
---|---|---
calculate|aca|2.1.0
vscode-markdownlint|Dav|0.31.0
path-autocomplete|ion|1.13.3
team|ms-|1.149.2
vscode-table-formatter|shu|1.2.1
vscode-autohotkey|sle|0.2.2
vscode-todo-highlight|way|1.0.4
markdown-all-in-one|yzh|2.5.1

(1 theme extensions excluded)

</details>

**Comment by pavangayakwad** [comment]
Yes, happening this to me too!!

Version: 1.39.1 (user setup)
Commit: 88f15d1
Date: 2019-10-10T23:31:28.683Z
Electron: 4.2.10
Chrome: 69.0.3497.128
Node.js: 10.11.0
V8: 6.9.427.31-electron.0
OS: Windows_NT x64 10.0.18362

**Comment by daFritz84** [comment]
Same

Version: 1.39.2 (system setup)
Commit: 6ab598523be7a800d7f3eb4d92d7ab9a66069390
Date: 2019-10-15T26:35:18.241Z
Electron: 4.2.10
Chrome: 69.0.3497.128
Node.js: 10.11.0
V8: 6.9.427.31-electron.0
OS: Windows_NT x64 6.1.7601

**Comment by alexandrudima** [comment]
@tobiaslombacher @pavangayakwad @daFritz84 Does this reproduce when disabling all extensions?

**Comment by pavangayakwad** [comment]
Well, I uninstalled completely, restarted my machine and then installed freshly, Not getting this issue since from 2 days of my regular work. Yeah, all my extensions are lost and I installed few of them which I use regularly.

**Comment by daFritz84** [comment]
> @tobiaslombacher @pavangayakwad @daFritz84 Does this reproduce when disabling all extensions?
> 
> Can you please share your installed extensions? Perhaps there is an extension common between the three of you that might cause the crash of the extension host process?

Things I tried:
 - disabled all extensions with  the command line switch
 - uninstalled all extensions 
 - tried to use portable mode (zip file with data directory)
So, I'm quite sure it has something to do with my particular windows version / configuration.

However, I did not try @pavangayakwad method yet.

**Comment by pavangayakwad** [comment]
OK here is the list of my Extensions for now. However, lost others when I uninstalled. 

Not sure how much this would help you now, however, sharing it anyway.

VS Code version: Code 1.39.2 (6ab598523be7a800d7f3eb4d92d7ab9a66069390, 2019-10-15T15:35:18.241Z)
OS version: Windows_NT x64 10.0.18362

<details>
<summary>System Info</summary>

|Item|Value|
|---|---|
|CPUs|Intel(R) Core(TM) i3-4005U CPU @ 1.70GHz (4 x 1696)|
|GPU Status|2d_canvas: enabled<br>flash_3d: enabled<br>flash_stage3d: enabled<br>flash_stage3d_baseline: enabled<br>gpu_compositing: enabled<br>multiple_raster_threads: enabled_on<br>native_gpu_memory_buffers: disabled_software<br>oop_rasterization: disabled_off<br>protected_video_decode: unavailable_off<br>rasterization: enabled<br>skia_deferred_display_list: disabled_off<br>skia_renderer: disabled_off<br>surface_synchronization: enabled_on<br>video_decode: enabled<br>viz_display_compositor: disabled_off<br>webgl: enabled<br>webgl2: enabled|
|Load (avg)|undefined|
|Memory (System)|7.93GB (2.90GB free)|
|Process Argv||
|Screen Reader|no|
|VM|0%|
</details><details><summary>Extensions (52)</summary>

Extension|Author (truncated)|Version
---|---|---
vscode-tomcat|ada|0.11.1
ng-template|Ang|0.802.3
path-intellisense|chr|1.4.2
dart-code|Dar|3.5.1
flutter|Dar|3.5.1
vscode-eslint|dba|1.9.1
FreeMarker|dco|0.0.9
xml|Dot|2.5.0
EditorConfig|Edi|0.14.2
vscode-npm-script|eg2|0.3.9
prettier-vscode|esb|2.3.0
php-debug|fel|1.13.0
code-runner|for|0.9.14
svgeditor|hen|2.9.0
beautify|Hoo|1.5.0
phpcs|ika|1.0.5
angular-essentials|joh|0.6.3
Angular2|joh|8.1.1
thymeleaf|juh|1.0.3
vscode-JS-CSS-HTML-formatter|lon|0.2.3
HTMLHint|mka|0.6.0
hg|mrc|1.3.0
vscode-azurefunctions|ms-|0.19.1
mssql|ms-|1.6.0
python|ms-|2019.10.41019
azure-account|ms-|0.8.6
csharp|ms-|1.21.5
Go|ms-|0.11.7
debugger-for-chrome|msj|4.12.0
angular2-inline|nat|0.0.17
vetur|oct|0.22.4
vscode-boot-dev-pack|Piv|0.0.8
vscode-concourse|Piv|1.11.0
vscode-manifest-yaml|Piv|1.11.0
vscode-spring-boot|Piv|1.11.0
material-icon-theme|PKi|3.9.1
vscode-jsp|pth|0.0.2
java|red|0.51.0
vscode-odata|sta|0.1.0
bracket-padder|via|0.1.1
vscodeintellicode|Vis|1.1.9
vscode-java-debug|vsc|0.22.0
vscode-java-dependency|vsc|0.6.0
vscode-java-pack|vsc|0.8.0
vscode-java-test|vsc|0.20.0
vscode-maven|vsc|0.19.1
vscode-spring-boot-dashboard|vsc|0.1.6
vscode-spring-initializr|vsc|0.4.6
JavaScriptSnippets|xab|1.7.2
vscode-java-copilot|yao|0.1.4
vscode-java-saber|You|0.1.0
html-css-class-completion|Zig|1.19.0

(4 theme extensions excluded)

</details>

**Comment by tobiaslombacher** [comment]
I am using usually a portable installation. 

I have 
- removed all extensions on existing installs and 
- tried clean new portable installations with and without the extensions or user data.

I found out (all tried with the portable version, though I have tested 1.39.1 also with the user install and system install)
- 1.37.1 - with or without extensions, running in the user account is working
- 1.38.1 and .2 - with or without extensions, running in the **user** account is **not** working
- 1.38.1 and .2 - with or without extensions, running in the **local admin** account **is** working without crashing the extension host.
- 1.39.1 and .2 - behave the same as 1.38.1 and .2

It seems to be independent of the extensions in use since it crashes also without any extensions installed or enabled.

**Comment by daFritz84** [comment]
Just for the record:
I tried uninstalling 1.37.1, restarting and installing with 1.39.2 without luck.

**Comment by alexandrudima** [comment]
@tobiaslombacher Does anything appear in the developer tools (`F1 > Developer: Toggle Developer Tools`) when the crash occurs? Can you share the extension host / window logs? (`F1 > Developer: Open Logs Folder...`)

@bpasero Would you have any idea how to troubleshoot this further?

**Comment by bpasero** [comment]
@alexandrudima possibly with the Debug Diagnostics Tools on Windows as outlined at https://github.com/Microsoft/vscode/wiki/Native-Crash-Issues#windows. However this would require that the user has time enough to attach to the process.

Another thing maybe worth trying is to see if this reproduces out of sources. If so, we could add a bit of delay before starting the extension host so that this tool can be attached.

**Comment by bpasero** [comment]
Also now that we are shipping insiders on E6, possibly a crash report is generated? Maybe about time to push https://github.com/microsoft/vscode/issues/80858 so that we can test this on users :)

**Comment by tobiaslombacher** [comment]
Here are my log files.

From the console window:
[-1571463428257.log](https://github.com/microsoft/vscode/files/3746242/-1571463428257.log)
other logs:
[main.log](https://github.com/microsoft/vscode/files/3746239/main.log)
[exthost.log](https://github.com/microsoft/vscode/files/3746240/exthost.log)

If I need to get more details please let me know.

**Comment by tobiaslombacher** [comment]
I have captured the logs with the Debug Diagnostic Tool.

When the extension host terminated, I restarted it 3 times. 

The logs are attached.
[Code__PID__9984__Date__10_19_2019__Time_09_14_17AM__105__Log.txt](https://github.com/microsoft/vscode/files/3746310/Code__PID__9984__Date__10_19_2019__Time_09_14_17AM__105__Log.txt)
[Code__PID__21876__Date__10_19_2019__Time_09_14_12AM__785__Log.txt](https://github.com/microsoft/vscode/files/3746311/Code__PID__21876__Date__10_19_2019__Time_09_14_12AM__785__Log.txt)
[Code__PID__5948__Date__10_19_2019__Time_09_14_12AM__955__Log.txt](https://github.com/microsoft/vscode/files/3746312/Code__PID__5948__Date__10_19_2019__Time_09_14_12AM__955__Log.txt)
[Code__PID__14804__Date__10_19_2019__Time_09_14_12AM__874__Log.txt](https://github.com/microsoft/vscode/files/3746313/Code__PID__14804__Date__10_19_2019__Time_09_14_12AM__874__Log.txt)
[Code__PID__18272__Date__10_19_2019__Time_09_14_17AM__194__Log.txt](https://github.com/microsoft/vscode/files/3746314/Code__PID__18272__Date__10_19_2019__Time_09_14_17AM__194__Log.txt)
[Code__PID__21080__Date__10_19_2019__Time_09_14_36AM__858__Log.txt](https://github.com/microsoft/vscode/files/3746315/Code__PID__21080__Date__10_19_2019__Time_09_14_36AM__858__Log.txt)
[Code__PID__10216__Date__10_19_2019__Time_09_14_23AM__826__Log.txt](https://github.com/microsoft/vscode/files/3746316/Code__PID__10216__Date__10_19_2019__Time_09_14_23AM__826__Log.txt)
[Code__PID__3684__Date__10_19_2019__Time_09_14_27AM__830__Log.txt](https://github.com/microsoft/vscode/files/3746317/Code__PID__3684__Date__10_19_2019__Time_09_14_27AM__830__Log.txt)
[Code__PID__21364__Date__10_19_2019__Time_09_14_17AM__10__Log.txt](https://github.com/microsoft/vscode/files/3746318/Code__PID__21364__Date__10_19_2019__Time_09_14_17AM__10__Log.txt)

**Comment by bpasero** [comment]
Interesting one:

```
***********************
*  EXCEPTION DETAILS  *
***********************

DetailID = 1
	Count:    2
	Exception #:  0XC0000005
	Stack:        
		CRYPT32!CertCloseStore+0x2fe
		CRYPT32!CertCloseStore+0x2e3
		CRYPT32!CertEnumCertificatesInStore+0x1f
		crypt32_7fed42e0000+0x1ca9
		crypt32_7fed42e0000+0x338d
		Code!napi_create_function+0x34a
		Code!v8::Extension::~Extension+0xc1c
		Code!v8::ScriptCompiler::ExternalSourceStream::ResetToBookmark+0x291cf
		Code!v8::ScriptCompiler::ExternalSourceStream::ResetToBookmark+0x28751
		Code!v8::ScriptCompiler::ExternalSourceStream::ResetToBookmark+0x283d1
		Code!v8::EmbedderHeapTracer::FinalizeTracing+0x23aa62
		0x00000040`b10bb801
		0x00000209`10d3fe39
		0x00000209`10d3f309
		0x00000078`08788949
		0x00000008`00000000
		0x00000159`d0f05751
		0x00000209`10d56371
		0x0000014d`dad19081
		0x2bdd00
		0x6
		0x2bde20
		0x0000014d`dad0880d
		0x00000159`d0f026f1
		0x00000078`0878dd31
		0x00000005`00000000
		0x00000159`d0f02801
		0x00000209`10d56371
		0x00000078`0878dd31
		0x00000209`10d3f309
```

**Comment by alexandrudima** [comment]
Thanks a lot @tobiaslombacher and @bpasero !

---

@chrmarti This appears to be an access violation in `vscode-windows-ca-certs`, specifically [on this line](https://github.com/microsoft/vscode-windows-ca-certs/blob/9e9e73f1d11db2d8f08292443f1c38fb1bbeead9/crypt32.cc#L43)

---

From `Code__PID__21080__Date__10_19_2019__Time_09_14_36AM__858__Log.txt`:

Notice `crypt32.node` getting loaded from `vscode-windows-ca-certs`:
```
  \\?\c:\Users\lot\Downloads\vscode1392\resources\app\node_modules.asar.unpacked\vscode-windows-ca-certs\build\Release\crypt32.node loaded at 0xd42e0000
```

Notice the access violation top stack points to `CRYPT32`
```
DetailID = 1
	Count:    2
	Exception #:  0XC0000005
	Stack:        
		CRYPT32!CertCloseStore+0x2fe
		CRYPT32!CertCloseStore+0x2e3
		CRYPT32!CertEnumCertificatesInStore+0x1f
		crypt32_7fed42e0000+0x1ca9
		crypt32_7fed42e0000+0x338d
		Code!napi_create_function+0x34a
		Code!v8::Extension::~Extension+0xc1c
		Code!v8::ScriptCompiler::ExternalSourceStream::ResetToBookmark+0x291cf
		Code!v8::ScriptCompiler::ExternalSourceStream::ResetToBookmark+0x28751
		Code!v8::ScriptCompiler::ExternalSourceStream::ResetToBookmark+0x283d1
		Code!v8::EmbedderHeapTracer::FinalizeTracing+0x23aa62
```

**Comment by tobiaslombacher** [comment]
Do we need to change anything in the group policies? Does anybody know what needs to be changed?

Somehow it is strange that this happened only from 1.38.1 onwards and I am pretty sure that our GPO's did not change during this period.

**Comment by chrmarti** [comment]
You can try disabling `System Certificates` as a workaround:
![image](https://user-images.githubusercontent.com/9205389/67477868-3c8f6c80-f65b-11e9-8a6c-58acabde4e6d.png)

There was a bug before 1.38 preventing it from working and likely preventing it from crashing in your case too.

**Comment by daFritz84** [comment]
@chrmarti  yep, the workaround does the trick for me.

**Comment by chrmarti** [comment]
@tobiaslombacher and @daFritz84 I have pushed a potential fix, could you verify you can run with System Certificates enabled with tomorrow's (Friday's) Insiders build once it's out: https://code.visualstudio.com/insiders/

**Comment by tobiaslombacher** [comment]
**@chrmati** 
- disabling the `System Certificates` worked :+1: 
- tried 1.40 insiders - no success. The extension host still terminates, see below

> abstractExtensionService.ts:155 Extension host terminated unexpectedly. Code:  3221225477  Signal:  null
_onExtensionHostCrashed @ abstractExtensionService.ts:155
notificationsAlerts.ts:40 Extension host terminated unexpectedly.

**Comment by chrmarti** [comment]
@tobiaslombacher 

Could you install Node.js 12 and try in an empty folder:
- `npm i vscode-windows-ca-certs`
- Create a `test.js` file:
```js
console.log('Node: ' + JSON.stringify(process.versions));
 
console.log('Loading');
const winCA = require('vscode-windows-ca-certs');
 
let ders = [];
console.log('Opening');
const store = winCA();
try {
    let der;
    console.log('Reading');
    while (der = store.next()) {
        ders.push(der);
    }
    console.log('Read ' + ders.length);
} finally {
    console.log('Closing');
    store.done();
}
console.log('Done');
```
- `node test.js`

And attach the output of that?

**Comment by alexandrudima** [comment]
@tobiaslombacher Can you please try again with today's Insiders build? We had some test failures on Friday and the Friday insiders build was released a bit delayed.

**Comment by tobiaslombacher** [comment]
**@alexandrudima** I have tried with today's build 1.40.0. Now it is working without problems. 

**@chrmarti** I will have to get it installed and running first. Give me a little time. Not so easy if you are in a domain network.

**Comment by chrmarti** [comment]
@tobiaslombacher No need to try that if the Insiders build works. Thanks.

@daFritz84 Could you give the Insiders build a try and check if it works with System Certificates enabled too?

**Comment by daFritz84** [comment]
@chrmarti A little late to the party, but I can also confirm that the latest Insider Build fixes this issue

**Comment by thanhnguyen92** [comment]
@daFritz84 new download from website still have issue:

`Version: 1.40.1 (user setup)
Commit: 8795a9889db74563ddd43eb0a897a2384129a619
Date: 2019-11-13T16:49:35.976Z
Electron: 6.1.2
Chrome: 76.0.3809.146
Node.js: 12.4.0
V8: 7.6.303.31-electron.0
OS: Windows_NT x64 10.0.17763`

**Comment by daFritz84** [comment]
@thanhnguyen92
I'm running 1.40.1 (user setup) on Windows 10 without any troubles.

You might wanna try some of the debugging steps listed in this thread, like disabling all extensions or doing some [native debugging](https://github.com/Microsoft/vscode/wiki/Native-Crash-Issues#windows). 

In any case, opening a new issue might be appropriate.

**Comment by thanhnguyen92** [comment]
@daFritz84 of course I did all above steps:
- Disable `System Certificates`
- Uninstall/install Visual Code
- Remove all extensions

**Comment by daFritz84** [comment]
@thanhnguyen92 
Then you might wanna create a new issue referencing this one. I cannot reopen this issue since I'm neither the original thread creator nor an admin in this project.

**Comment by thanhnguyen92** [comment]
> @thanhnguyen92
> Then you might wanna create a new issue referencing this one. I cannot reopen this issue since I'm neither the original thread creator nor an admin in this project.

Thanks @daFritz84 , I've created the new one #85586
